virtual_rainforest/core/base_model.py

```diff
@@ -517,3 +523,57 @@ def check_init_data(self) -> None:
             )
             LOGGER.error(error)
             raise error
+
+
+def register_model(module_name: str, schema_filename: str) -> None:
+    """Helper function to register the things required to setup a given model.
+
+    This registers the model configuration schema to the
+    :data:`~virtual_rainforest.core.config.SCHEMA_REGISTRY`. It also finds all constants
+    classes that have been defined and registers them to the
+    :data:`~virtual_rainforest.core.constants.CONSTANTS_REGISTRY`.
+
+    Args:
+        module_name: The name of the module containing the model to be registered
+        schema_filename: The filename of the model configuration schema
+
+    Raises:
+        ConfigurationError: If the module does not define a single Model class which is
+            a child of BaseModel.
+    """
+
+    # Get a reference to the module
+    module = sys.modules[module_name]
+
+    # Locate the BaseModel subclass for the model, which should be imported into the
+    # module root for the model
+    model_obj_found = [
+        obj
+        for nm, obj in getmembers(module)
+        if isclass(obj) and issubclass(obj, BaseModel)
+    ]
```

Comment:
Rather than doing this magic, why not just have the user pass in the model class directly? I think that would be clearer too, rather than importing the model classes into `__init__.py` then not using them. Explicit is better than implicit.